File: Bubble sort.py
__author__ = 'dongpei'
# not complete



def bubble_sort (alist):
    for passnum in range(len(alist)-1,0,-1):
        for i in range(passnum):
            if alist[i] > alist[i+1]:
                temp = alist[i+1]
                alist[i+1] = alist[i]
                alist[i] = temp

    return(alist)

alist = [54,26,93,17,77,31,44,55,20]
bubble_sort(alist)

File: Bubble.sort.R
# bubble sort

bubble.sort <- function(vec){
  for (passnum in (length(vector)-1):1) {
    for (i in 1:passnum){
      if (vec[i] > vec[i+1]){
        temp <- vec[i+1]
        vec[i+1] <- vec[i]
        vec[i] <- temp
      }
    }
  }
  return (vec)
}

vec <- c(54,26,93,17,77,31,44,55,20)
bubble.sort(vec)